### PR TITLE
Major Cargo Console Menu Revork

### PR DESCRIPTION
**Pull request by VlaDOS1408**
## About the PR

Subdivision of: https://github.com/space-wizards/space-station-14/pull/33632 pr
I have greatly improved the cargo order console. Now it can be conveniently used in Russian localization.

## Why / Balance

I just looked at this terrible window that was there before and realized that it needed to be changed. In Russian localization, it is simply difficult to use this, since the product line went far beyond the edges of the screen.

## Technical details

I have slightly changed the code of these windows so that everything works as it should, but I have not made any fundamental changes. I renamed the localization keys a bit, brought them to a common style.

### After
UIUpdate_CargoConsole
UIUpdate_CargoConsole-Huge
UIUpdate_CargoConsole-Low

And also, if the customer's field is left empty, it will be replaced with "Unknown"

## Requirements

- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [x] I have added media to this PR or it does not require an ingame showcase.
- [x] I put ticks on the pre-prepared places under the ticks.

**Changelog**

:cl:
- tweak: Cargo console UI reworked
- fix: fix Cargo console UI on another languages (root change for this)

### CODE REVIEW

File: Resources/Locale/en-US/cargo/cargo-console-component.ftl [hunk 1]
```diff
@@ -1,25 +1,32 @@
...
-cargo-console-menu-populate-orders-cargo-order-row-product-name-text = {$productName} (x{$orderAmount}) by {$orderRequester}
-cargo-console-menu-cargo-order-row-approve-button = Approve
-cargo-console-menu-cargo-order-row-cancel-button = Cancel
+cargo-console-menu-order-row-title = Order from {$orderRequester}
+cargo-console-menu-order-row-product-name = Order {$productName} (x{$orderAmount})
+cargo-console-menu-order-row-product-description = Reason: {$orderReason}
+cargo-console-menu-order-row-button-approve = Approve
+cargo-console-menu-order-row-button-cancel = Cancel
+cargo-console-menu-order-row-alerts-reason-absent = The reason is not specified
+cargo-console-menu-order-row-alerts-requester-unknow = Unknow
```
Comment (Aeshus) [inline Resources/Locale/en-US/cargo/cargo-console-component.ftl:29]:
```suggestion
cargo-console-menu-order-row-alerts-requester-unknown = Unknown
```

File: Content.Client/Cargo/BUI/CargoOrderConsoleBoundUserInterface.cs [hunk 2]
```diff
@@ -68,7 +68,7 @@ protected override void Open()
 
             _menu.OnItemSelected += (args) =>
             {
-                if (args.Button.Parent is not CargoProductRow row)
+                if (args.Button.Parent?.Parent is not CargoProductRow row)
```
Comment (Aeshus) [inline Content.Client/Cargo/BUI/CargoOrderConsoleBoundUserInterface.cs:71]: Is there any better way of doing this? It feels like a bad code smell to have to rely on the linked-lists of ui nodes.

File: Content.Client/Cargo/UI/CargoConsoleMenu.xaml [hunk 3]
```diff
@@ -1,68 +1,200 @@
...
+                                </BoxContainer>
+                            </ScrollContainer>
+                        </BoxContainer>
+                    </PanelContainer>
+                </BoxContainer>
+            </BoxContainer>
+        </BoxContainer>
+
+        <!-- Footer -->
+        <!-- TODO: Create customControls element -->
```
Comment (Aeshus) [inline Content.Client/Cargo/UI/CargoConsoleMenu.xaml:171]: Is this todo still relevant?

File: Content.Client/Cargo/UI/CargoConsoleMenu.xaml.cs [hunk 4]
```diff
@@ -1,3 +1,4 @@
+using System;
```
Comment (Aeshus) [inline Content.Client/Cargo/UI/CargoConsoleMenu.xaml.cs:1]: Import not used.
```suggestion
```

File: Content.Client/Cargo/UI/CargoConsoleOrderMenu.xaml [hunk 5]
```diff
@@ -1,33 +1,60 @@
...
-                     Value="1" />
+                Access="Public"
+                HorizontalExpand="True"
+                Value="1" />
+
         </GridContainer>
+
+        <Control VerticalExpand="True"/>
+
+
```
Comment (Aeshus) [inline Content.Client/Cargo/UI/CargoConsoleOrderMenu.xaml:52]: Extra spacing
```suggestion
```

**Review by Aeshus — CHANGES_REQUESTED** [review]

File: Content.Client/Cargo/BUI/CargoOrderConsoleBoundUserInterface.cs (hunk 2 above)
Comment (VlaDOS1408) [reply Content.Client/Cargo/BUI/CargoOrderConsoleBoundUserInterface.cs:71]: Let's fix all this without me, if you need, and I'll stand aside. I haven't changed anything in principle, so you can assume that I didn't touch this file.

**Comment by Olexiy-Lapin** [comment]
Красота

**Comment by TheShuEd** [comment]
Due to the author being banned from the repository, we will be closing this PR. Other contributors may continue the work by opening up a new PR. Should the author successfully appeal the ban, the PR may be re-opened.

**Pull request closed without merging**